quill-core/src/main/scala/io/getquill/norm/BetaReduction.scala

```diff
@@ -8,7 +8,12 @@ case class BetaReduction(map: collection.Map[Ident, Ast])
   override def apply(ast: Ast) =
     ast match {
       case Property(Tuple(values), name) =>
-        apply(values(name.drop(1).toInt - 1))
+        val aliases = values.distinct
+        aliases match{
```

Comment:
Why do you need this transformation? Is it because the AST has two aliases and the new join type has only one? Maybe we should have a new AST class for the new type of join?